src/compiler/binder.ts

```diff
@@ -2502,7 +2524,11 @@ namespace ts {
                     // Bind this property to the containing class
                     const containingClass = thisContainer.parent;
                     const symbolTable = hasModifier(thisContainer, ModifierFlags.Static) ? containingClass.symbol.exports! : containingClass.symbol.members!;
-                    declareSymbol(symbolTable, containingClass.symbol, node, SymbolFlags.Property, SymbolFlags.None, /*isReplaceableByMethod*/ true);
+                    // private names *must* be declared (even in JS files)
+                    const hasPrivateName = isBinaryExpression(node) && isPropertyAccessExpression(node.left) && isPrivateIdentifier(node.left.name);

```

Comment:
It looks like it's checking that it's not an assignment to a private identifier property. I can do that for `PrivateIdentifierAssignmentExpression`, but I'd like to keep `PrivateIdentifierPropertyAccessExpression` because inlining it seems to require a cast. When I have a function that takes `PropertyAccessExpression & { name: PrivateIdentifier }`, it doesn't seem to work after I do: `isPropertyAccessExpression(node) && isPrivateIdentifier(node.name)`.

Reply:
@mheiber I removed the `isPrivateIdentifierAssignmentExpression` and corresponding interface (in favor of `isAssignmentExpression` + `isPrivateIdentifierPropertyAccessExpression`)